nasa/cumulus
# [CUMULUS-836] configurable docker storage driver

**Summary:** 

Addresses [CUMULUS-836: dm.basesize causes our docker image to fail a restart (using efs)](https://bugs.earthdata.nasa.gov/browse/CUMULUS-836)

## Changes

See CHANGELOG.md

## Test Plan

- [x] Adhoc testing

## Testing steps:

* Deploy [Cumulus Example stack](https://github.com/cumulus-nasa/cumulus/tree/master/example) with change to `deployment/app/config.yml` and `app/config.yml` of storageDriver and amiid respectively
* Bring up a new ECS instance to adopt changes by either killing the existing ECS instance and wait for autoscaling to create a new one or the deployment would start a new instance
* Run `sudo docker info` and verify storage is configured as expected
* Verify ECS cluster is listed as running a task on the new instance

Tested configurations:

|     | NGAP AMI (ami-a307b8dc)  | AWS ECS-optimized AMI (ami-a7a242da) |
| -------------------------------- | ---- | --- |
| `storageDriver: devicemapper`  |  ✔️  | ✔️ |
| `storageDriver: overlay2`            |  ✔️  | ✔️ |
